auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/GenericIdpActivity.kt

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.quickstart.auth.kotlin
+
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.FirebaseUser
+import com.google.firebase.auth.OAuthProvider
+import com.google.firebase.quickstart.auth.R
+import com.google.firebase.quickstart.auth.java.BaseActivity
+import kotlinx.android.synthetic.main.activity_generic_idp.detail
+import kotlinx.android.synthetic.main.activity_generic_idp.genericSignInButton
+import kotlinx.android.synthetic.main.activity_generic_idp.signOutButton
+import kotlinx.android.synthetic.main.activity_generic_idp.status
+
+import java.util.ArrayList
+
+/**
+ * Demonstrate Firebase Authentication using a Generic Identity Provider (IDP).
+ */
+class GenericIdpActivity : BaseActivity(), View.OnClickListener {
+
+    // [START declare_auth]
+    private lateinit var auth: FirebaseAuth
+    // [END declare_auth]
+
+    public override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_generic_idp)
+
+        // Initialize Firebase Auth
+        auth = FirebaseAuth.getInstance()
+
+        // Set up button click listeners
+        genericSignInButton.setOnClickListener(this)
+        signOutButton.setOnClickListener(this)
+    }
+
+    public override fun onStart() {
+        super.onStart()
+        // Check if user is signed in (non-null) and update UI accordingly.
+        val currentUser = auth.currentUser
+        updateUI(currentUser)
+
+        // Look for a pending auth result
+        val pending = auth.pendingAuthResult
+        if (pending != null) {
+            pending.addOnSuccessListener {
```

Comment:
Can we have the argument in the same line as the opening brace?
```suggestion
            pending.addOnSuccessListener { authResult ->
```